test-app/tests/integration/components/combobox-test.gts

```diff
@@ -0,0 +1,854 @@
+    assert
+      .dom('[role="option"]:last-child')
+      .hasAttribute('data-active', 'true');
+  });
+
+  test('it sets the "active" item to the previous item in the list when using the UP arrow', async function (assert) {
+    // NOTE: Setting the selected option to "red" here so that
+    // the last item in the list will be active so that we can
+    // move up!
+    await render(<template>
+      <ComboboxControl
+        @selected="red"
+        @options={{testColors}}
+        data-combobox
+        as |combobox|
+      >
+        <combobox.Option>{{combobox.option}}</combobox.Option>
+      </ComboboxControl>
+    </template>);
+
+    await click('[data-combobox]');
+
+    assert.dom('[role="option"]').exists({ count: 2 });
+
+    await triggerKeyEvent('[data-combobox]', 'keydown', 'ArrowUp');
+
+    assert
+      .dom('[role="option"]:first-child')
+      .hasAttribute('data-active', 'true');
+    assert
+      .dom('[role="option"]:last-child')
+      .hasAttribute('data-active', 'false');
+  });
+
+  test('it closes an open popover when the ESCAPE key is pressed', async function (assert) {
+    await render(<template>
+      <ComboboxControl @options={{testColors}} data-combobox as |combobox|>
+        <combobox.Option>{{combobox.option}}</combobox.Option>
+      </ComboboxControl>
+    </template>);
+
+    await click('[data-combobox]');
+
+    assert.dom('[role="listbox"]').exists();
+
+    await triggerKeyEvent('[data-combobox]', 'keydown', 'Escape');
+
+    assert.dom('[role="listbox"]').doesNotExist();
+  });
+
+  test('it closes an open popover when the component is blurred', async function (assert) {
+    await render(<template>
+      {{! template-lint-disable require-input-label }}
+      <input placeholder="test" data-input />
+
+      <ComboboxControl @options={{testColors}} data-combobox as |combobox|>
+        <combobox.Option>{{combobox.option}}</combobox.Option>
+      </ComboboxControl>
+    </template>);
+
+    // Open the popover
+    await click('[data-combobox]');
+
+    assert.dom('[role="listbox"]').exists();
+
+    // Now blur the elment
+    await click('[data-input]');
+
+    assert.dom('[role="listbox"]').doesNotExist();
+  });
+
+  test('it reopens the popover when any key is pressed if the popover is closed', async function (assert) {
+    await render(<template>
+      <ComboboxControl @options={{testColors}} data-combobox as |combobox|>
+        <combobox.Option>{{combobox.option}}</combobox.Option>
+      </ComboboxControl>
+    </template>);
+
+    // Open the popover by clicking it
+    await click('[data-combobox]');
+
+    // Now close it
+    await triggerKeyEvent('[data-combobox]', 'keydown', 'Escape');
+    // Now reopen it
+    await triggerKeyEvent('[data-combobox]', 'keydown', 'ArrowDown');
+
+    assert.dom('[role="listbox"]').exists();
+  });
+
+  test('it makes the first option "active" when the metakey and UP arrow is pressed', async function (assert) {
```

Comment:
Might be worth pointing out that metakey is CMD in Mac (not sure what it is in Windows, CTRL?)

Reply:
also `active` is the "currently hovered" one in this case? you still have to press ENTER to select? (just to clarify)

Reply:
> also active is the "currently hovered" one in this case? you still have to press ENTER to select? (just to clarify)

Correct!